File: scholar-business/src/main/java/com/varun/scholar/business/data/entities1/SubjectAllocation.java
package com.varun.scholar.business.data.entities1;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.varun.scholar.business.data.entities.AbstractEntity;

@Entity
public class SubjectAllocation extends AbstractEntity {
	String batchName;
	String courseName;
	String section;
	String subject;
	boolean supervisor;
	@Temporal(TemporalType.DATE)
	Date startDate;
	@Temporal(TemporalType.DATE)
	Date endDate;
	@Temporal(TemporalType.DATE)
	Date lastChange;
}

File: scholar-web/src/main/webapp/app/model/staff/lookup/SubjectAllocation.js
Ext.define('scholar.model.staff.lookup.SubjectAllocation', {
	extend: 'Ext.data.Model',
	fields : [ {
		name : 'id',
		type : 'int'
	}, {
		name : 'batchName',
		type : 'string'
	}, {
		name : 'courseName',
		type : 'string'
	}, {
		name : 'section',
		type : 'string'
	},
	{
		name : 'subject',
		type : 'string'
	},
	{
		name : 'supervisor',
		type : 'bool'
	},
	{
		name : 'startDate',
		type : 'date',
		dateFormat : 'n/j h:ia'
	}, {
		name : 'endDate',
		type : 'date',
		dateFormat : 'n/j h:ia'
	}, {
		name : 'lastChange',
		type : 'date',
		dateFormat : 'n/j h:ia'
	} ]
	
});